cmd/commands/cli/command.go

```diff
@@ -38,29 +39,31 @@ const cliCommandName = "cli"
 // NewCommand constructs CLI based Mysterium UI with possibility to control quiting
 func NewCommand() *cli.Command {
 	var di cmd.Dependencies
+	var cmdCLI cliApp
 
 	return &cli.Command{
 		Name:  cliCommandName,
 		Usage: "Starts a CLI client with a Tequilapi",
 		Action: func(ctx *cli.Context) error {
-			errorChannel := make(chan error)
+			errorChannel := make(chan error, 3)
 			if err := di.Bootstrap(cmd.ParseFlagsNode(ctx)); err != nil {
 				return err
 			}
 			go func() { errorChannel <- di.Node.Wait() }()
 
 			nodeOptions := cmd.ParseFlagsNode(ctx)
-			cmdCLI := &cliApp{
-				historyFile: filepath.Join(nodeOptions.Directories.Data, ".cli_history"),
-				tequilapi:   tequilapi_client.NewClient(nodeOptions.TequilapiAddress, nodeOptions.TequilapiPort),
-			}
+			cmdCLI.historyFile = filepath.Join(nodeOptions.Directories.Data, ".cli_history")
+			cmdCLI.tequilapi = tequilapi_client.NewClient(nodeOptions.TequilapiAddress, nodeOptions.TequilapiPort)
+
 			go func() { errorChannel <- cmdCLI.Run() }()
-			cmd.RegisterSignalCallback(utils.SoftKiller(cmdCLI.Kill))
-			cmd.RegisterSignalCallback(utils.SoftKiller(di.Shutdown))
+			cmd.RegisterSignalCallback(func() { errorChannel <- nil })
```

Comment:
Did You alligned change to this stop flow also:
`tequilapi_endpoints.AddRouteForStop(router, utils.SoftKiller(di.Shutdown))`

Reply:
Didn't get a question. 

In the [master](https://github.com/mysteriumnetwork/node/blob/master/core/node/node.go#L117) I see 
```
utils.SoftKiller(node.Kill)
```

So it will stop the `node` and service will be exited correctly, and `di.Shutdown()` will be applies from `After` section.